src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -686,10 +653,197 @@ public void afterIndexShardStarted(IndexShard indexShard) {
                 threadPool.generic().execute(() -> {
                     if (isSecurityIndexRestoredFromSnapshot(clusterService, index, securityIndex)) {
                         LOGGER.info("Security index primary shard {} started - config reloading for snapshot restore", shardId);
-                        reloadConfiguration(CType.values());
+                        reloadConfiguration(CType.values(), null);
                     }
                 });
             }
         }
     }
+
+    /**
+     * This class is responsible for managing requests to reload the security index. Its main purpose
+     * is to make sure that there is no unbounded queue of reload requests. Rather, it works this way:
+     * <ul>
+     *     <li>If there is no reload activity, just schedule it immediately.</li>
+     *     <li>If a reload is currently in process, schedule a further reload right afterwards.</li>
+     *     <li>If a reload is currently in process and a further reload is already scheduled, just rely on the already scheduled reload.
```

Comment:
there will only be at max 2 reloads in queue, is that correct or will it depend on the configuration type?

Reply:
There is always max 1 reload active and one further queued, which however might have a growing set of config types to be reloaded.